### PR TITLE
Update 'IC' to 'ICP'

### PR DESCRIPTION
**Pull request by jessiemongeon1**
This PR updates all instances of 'IC' to 'ICP' in the Dev Docs.

Two exceptions to this update are references to the "IC SDK" and "IC Specification".

### CODE REVIEW

**Review by MarioDfinity — APPROVED** [review]

File: docs/concepts/what-is-ic.md [hunk 1]
```diff
@@ -66,19 +66,19 @@ The Internet Computer also enables developers to use [cryptographically-secure i
...
+- Get started developing on ICP by embarking on your [developer journey](/docs/current/tutorials/developer-journey).
 
 - [Designing a dapp](/docs/current/developer-docs/backend/design-dapps).
 
 - [Building backend canisters with Motoko](/docs/current/developer-docs/backend/motoko/).
 
 - [Building backend canisters with Rust](/docs/current/developer-docs/backend/rust/).
 
-## Connect with the IC community
+## Connect with ICP community
```
Comment (jessiemongeon1) [inline docs/concepts/what-is-ic.md:79]:
```suggestion
## Connect with the ICP community
```

File: docs/developer-docs/agents/index.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
-# Integrating external agents with the IC
+# Integrating external agents with ICP
 
 ## Overview
 
-In the Internet Computer ecosystem, a library that is used to make calls to the IC public interface is called an **agent**. An agent has a few key responsibilities, which make it convenient to work with in your language of choice.
+In the Internet Computer ecosystem, a library that is used to make calls to ICP public interface is called an **agent**. An agent has a few key responsibilities, which make it convenient to work with in your language of choice.
```
Comment (jessiemongeon1) [inline docs/developer-docs/agents/index.md:5]:
```suggestion
In the Internet Computer ecosystem, a library that is used to make calls to the ICP public interface is called an **agent**. An agent has a few key responsibilities, which make it convenient to work with in your language of choice.
```

File: docs/developer-docs/agents/index.md [hunk 3]
```diff
@@ -53,21 +53,21 @@ A `call` to the Internet Computer can take two common forms - an `update` or a `
...
+Once the data has been returned from ICP, the **agent** takes the certificate from the payload and verifies it. The certificate can be verified as genuine using the public `rootKey` of the NNS subnet. The network will respond with a CBOR-encoded buffer, which the **agent** can then decode, and transform into a useful structure using semantic language-specific types. For example, if the type returned from the canister is `text`, that will get turned into a JavaScript `string`, and so on.
 
 ### Managing authentication
 
 Calls to the Internet Computer always need to have a cryptographic identity attached. That identity will either be **anonymous** or **authenticated**, using a cryptographic signature. Since identities are required, canisters can use the identity attached to a call to decide how to respond to that call. This enables contracts to use those identities for other purposes.
 
 #### Accepted identities
 
-The IC accepts calls using the following types of signatures in identities:
+The ICP accepts calls using the following types of signatures in identities:
```
Comment (jessiemongeon1) [inline docs/developer-docs/agents/index.md:70]:
```suggestion
ICP accepts calls using the following types of signatures in identities:
```

File: docs/references/asset-canister.md [hunk 4]
```diff
@@ -155,7 +155,7 @@ Here is a sample:
       // Security: The Content Security Policy (CSP) given below aims at working with many apps rather than providing maximal security.
       // We recommend tightening the CSP for your specific application. Some recommendations are as follows:
       // - Use the CSP Evaluator (https://csp-evaluator.withgoogle.com/) to validate the CSP you define.
-      // - Follow the “Strict CSP” recommendations (https://csp.withgoogle.com/docs/strict-csp.html). However, note that in the context of the IC,
+      // - Follow the “Strict CSP” recommendations (https://csp.withgoogle.com/docs/strict-csp.html). However, note that in the context of ICP,
```
Comment (dfx-json) [inline docs/references/asset-canister.md:158]:
```suggestion
      // - Follow the “Strict CSP” recommendations (https://csp.withgoogle.com/docs/strict-csp.html). However, note that in the context of the mainnet (network=ic),
```

File: docs/developer-docs/setup/quickstart.md [hunk 5]
```diff
@@ -24,7 +24,7 @@ For more information on installing the IC SDK, please see [here](./install/index
 
 ## Start the local replica
 
-To start a local instance of the IC replica, use the command:
+To start a local instance of local replica, use the command:
```
Comment (dfx-json) [inline docs/developer-docs/setup/quickstart.md:27]:
```suggestion
To start a local replica instance, use the command:
```

File: docs/developer-docs/integrations/sns/testing/testing-on-mainnet.md [hunk 6]
```diff
@@ -127,7 +127,7 @@ Register all canisters that are supposed to be managed by the testflight SNS
 by submitting an SNS proposal via `quill`.
 
 When running the testflight locally, export the environment variable `IC_URL`
-to point to your local IC instance, e.g.,
+to point to your local ICP instance, e.g.,
```
Comment (dfx-json) [inline docs/developer-docs/integrations/sns/testing/testing-on-mainnet.md:130]:
```suggestion
to point to your local replica instance, e.g.,
```

File: docs/developer-docs/integrations/independently-verifying-ic-signatures.md [hunk 7]
```diff
@@ -2,17 +2,17 @@
 
 ## Overview
 
-Calls on the IC require that a cryptographic `signature` and `principal` are attached, usually in the form of an identity, be attached to the call. This identity can be anonymous or authenticated using a cryptographic signature. Canisters can use this attached identity to respond to a call, which enables contracts to use those identities for other workflows and purposes. 
+Calls on ICP require that a cryptographic `signature` and `principal` are attached, usually in the form of an identity, be attached to the call. This identity can be anonymous or authenticated using a cryptographic signature. Canisters can use this attached identity to respond to a call, which enables contracts to use those identities for other workflows and purposes.
```
Comment (dfx-json) [inline docs/developer-docs/integrations/independently-verifying-ic-signatures.md:5]:
```suggestion
Network calls on ICP require that a cryptographic `signature` and `principal` are attached, usually in the form of an identity, be attached to the call. This identity can be anonymous or authenticated using a cryptographic signature. Canisters can use this attached identity to respond to a call, which enables contracts to use those identities for other workflows and purposes. 
```

**Review by dfx-json — APPROVED** [review]

File: docs/developer-docs/agents/ic-agent-dfinity.md [hunk 8]
```diff
@@ -2,7 +2,7 @@
 sidebar_position: 4
 sidebar_label: "From a Rust agent"
 ---
-# Calling IC from a Rust agent
+# Calling ICP from a Rust agent
```
[the remark this answers is not in the document]
Comment (jessiemongeon1) [reply docs/developer-docs/agents/ic-agent-dfinity.md]: Updated to "Making a call from a Rust agent"

**Pull request merged**